### DISCUSSION TITLE
STIG Manger Login Screen Spinning After New Install

### DISCUSSION THREAD

**Discussion by mvharvey7** [NUWCDIVNPT/stig-manager#1308] [General]
I just took my first attempt installing Stig Manager. I am using the Docker nuwcdivnpt/stig-manager:1.2.7-ironbank, MYSQL 8 Image, and customer keycloak image. I am able to pull all files and run the docker-compose.yml file. Once I go to localhost:54000 I am presented with the username and password option. When I enter the credentials the Stig Manager just spins. Any idea of the issue. I have attached the install logs and screen shots including yml file.  Any help in identifying the issue would be greatly appreciated. 

One main error I do see is "mysql","type":"preflight","data":{"success":false,"message":"connect ECONNREFUSED 172.19.0.3:3306" I used the provided db:
    image: mysql:8.0
    ports:
      - "50001:3306"
    environment:
      - MYSQL_ROOT_PASSWORD=rootpw
      - MYSQL_USER=stigman
      - MYSQL_PASSWORD=stigman
      - MYSQL_DATABASE=stigman
    cap_add:
      - SYS_NICE  # workaround MySQL logging bug => mbind: Operation not permitted 

![SM-Screen](https://github.com/NUWCDIVNPT/stig-manager/assets/126341820/33ec6107-054e-4d4e-ae77-0862ff3fbba0)
[Install_Start_Output.txt](https://github.com/user-attachments/files/16018760/Install_Start_Output.txt)
[docker-compose-file.txt](https://github.com/user-attachments/files/16018776/docker-compose-file.txt)

**Comment by cd-rite** [comment]
Hi @mvharvey7 That `"mysql","type":"preflight","data":{"success":false,"message":"connect ECONNREFUSED 172.19.0.3:3306" ` message you see is expected at first, while the db is still coming up. Your log shows success: true later on for both the mysql and keycloak components.

Everything seems fine from the orchestration side....And the fact that you see a STIGMan screen at all means the API is happy and serving out the client.  If you refresh the page, do you still sit at that loading screen?  

You are logging in with one of the STIGMan usernames, not the keycloak admin console username, correct? ie. just "admin" and "password" once you are redirected to keycloak from the initial stigman client at  http://localhost:54000

**Reply by mvharvey7** [reply]
Thanks for the input on the 172.19.0.3 message. When I refresh the page, the page reloads, and it goes back to spinning. I am using Login using credentials "admin/password".  I can go to localhost:8080 in another tab and log into Keycloak with the provided password with no issues.

I attached new screen shots of the developer tools in edge of the errors. 

![SM-Login Screen](https://github.com/NUWCDIVNPT/stig-manager/assets/126341820/4b63ad4b-aafe-4636-ace8-c227643a135e)

![SM-Developer](https://github.com/NUWCDIVNPT/stig-manager/assets/126341820/d2494364-760a-48a8-817a-cdf53e19a880)

**Comment by csmig** [comment]
I can confirm the `nuwcdivnpt/stig-manager:latest` and `nuwcdivnpt/stig-manager:latest-ironbank` images published to Docker Hub via our Git Hub Action produce the misbehavior seen in the OP's dev tools. I have therefore removed those images and retagged the `:latest` images to be the same as the 1.4.11 images. Those images do not appear to have this problem. Please repull the desired image (`docker pull nuwcdivnpt/stig-manager:latest` or `docker pull nuwcdivnpt/stig-manager:latest-ironbank`) and retry.

FWIW, the OP said they were using Docker image `nuwcdivnpt/stig-manager:1.2.7-ironbank`, which is a very old release. However, the screenshot shows the current release version of 1.4.11, and the attached docker-compose.yml references the latest image, so I am assuming they are using `:latest`.

I've opened issue #1310 to track progress on fixing this.

**Comment by mvharvey7** [comment]
I repulled the image and now able to access the application. Thank you for your assistance!

![SM-Success](https://github.com/NUWCDIVNPT/stig-manager/assets/126341820/aca5c609-e68f-4d54-9cef-7ac13fc48448)
